BinaryToOctalConvertor.java

```diff
@@ -0,0 +1,59 @@
+/**
+ *  To convert binary numbers to octal
+ *  
+ *  version 1.0
+ *  @author Udit Saxena
+ *
+ */
+
+public class BinaryToOctalConvertor {
+	/** 
+	 * * This function convert binary to octal
+	 * first it convert binary to decimal then decimal to octal
+	 
+	 * @param n
+	 * @return octal value
+	 */
+	
+	
+ public int covertBinaryToOctal(int n) 
+ {
+	 
+	int checkBinary = n; //binary value
+	int digit=0;  //store digit value
+	while(checkBinary!=0)
+	{
+		digit=checkBinary%10; //for fetching single digit(last) from binary
+		if(digit!=0&&digit!=1) //check digit either 0 or 1
+		{
+			return -1;
+		}
+		checkBinary =checkBinary/10; //remove the use digit
+	}
+	
+	// Binary to decimal
```

Comment:
Binary to decimal and decimal to octal conversions can be extracted in separate methods.